### PR TITLE
feat: Option to download plan as JSON

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR introduces the option to download a plan as JSON, if a SQL or a binary plan was coded or uploaded.

### CODE REVIEW

File: client/src/components/SubstraitGraph.vue [hunk 1]
```diff
@@ -85,6 +99,16 @@ export default {
       };
       image.src = svgUrl;
     },
+    generateJSON() {
+      const jsonObject = JSON.parse(store.plan);
+      const jsonString = JSON.stringify(jsonObject, null, 4);
```
Comment (vibhatha) [inline client/src/components/SubstraitGraph.vue:104]: should we define `4` with a const variable and a proper definition for it (a comment would be nice)

Comment (sanjibansg) [reply client/src/components/SubstraitGraph.vue]: As discussed in PR #27, using `const` here will not be of much help. 
Commented on the usage of the `JSON.stringify()` parameters.

File: client/src/components/SubstraitGraph.vue [hunk 2]
```diff
@@ -85,6 +99,20 @@ export default {
       };
       image.src = svgUrl;
     },
+    generateJSON() {
+      const jsonObject = JSON.parse(store.plan);
+      const jsonString = JSON.stringify(
```
Comment (westonpace) [inline client/src/components/SubstraitGraph.vue:104]: Why parse it and then immediately stringify it?

File: client/src/views/CodeView.vue [hunk 3]
```diff
@@ -114,6 +117,7 @@ export default {
       const duckDbRsp = await axios.post("/api/parse/", {
         query: this.code,
       });
+      store.set_plan(duckDbRsp.data);
```
Comment (westonpace) [inline client/src/views/CodeView.vue:120]: Is there any harm in allowing download from JSON if the user uploaded from JSON?

**Pull request merged**